### DISCUSSION TITLE
List Issued Documents - How to filter results

### DISCUSSION THREAD

**Discussion by mesposito-bytecode** [fattureincloud/api#8] [Q&A] [answered]
Hi, I was looking at the API to retrieve documents.
It is not very clear to me how to use the query string parameter "q".
For example to search ei_data.od_number=number:
type=invoice
fields=ei_data
q=?

Thanks for your help

**Comment by mesposito-bytecode** [comment]
I found what I was looking for in the documentation.
https://developers.fattureincloud.it/docs/basics/additional-capabilities/filter-results

**Comment by valmoz** [comment, marked as answer]
Hi @mesposito-bytecode, I'm happy that you were able to solve it on your own.

The "q" parameter uses a SQL-like structure, that is described in the page that you linked. Feel free to contact us again if you need further info.

Mauro

**Reply by victoria-vj** [reply]
Hi Mauro,

You say that it's an SQL-like structure. I'm trying to create a filter query but are not able to figure out, how do you do the 'not contains'  operator to filter results that don't contain certain words?

Thank you.

Victoria

**Reply by valmoz** [reply]
Hi Victoria,
to be honest, I don't think we considered the "not contains" case.
By the way, I can check and let you know.

Mauro

**Reply by victoria-vj** [reply]
yes, please. That would be very useful. 🙂 

Thank you.

Victoria

**Reply by emajo** [reply]
Hi @victoria-vj , we checked and at the moment our APIs are not providing the possibility to filter with the `not contains` operator, but we will certainly think about adding it.

Thank you, 
Emanuele

**Reply by victoria-vj** [reply]
Hi Emanuele,

ok, thank you for checking.

Best regards,
Victoria
